### PR TITLE
[Task]: Add access level status to radio buttons

### PR DESCRIPTION
**Pull request by KobikahChandran**
## What this PR accomplishes

Shows the status of the selected radio button

## Issue(s) addressed
DATA-1404

## What needs review
- Shows correct status and colour for each radio button
- Filter facets show correctly in the right order
- Spacing/alignment on both desktop and mobile
- Status loads correctly when a button is selected

### CODE REVIEW

File: ckanext/ontario_theme/fanstatic/common.css [hunk 1]
```diff
@@ -3417,6 +3455,22 @@ label::after {
   z-index: 1
 }
 
+.access-level-status {
+  border-left: solid 5px;
+  padding: 24px;
+  background-color: #f2f2f2;
+  display: flex;
+  margin-left: 0.25rem;
+}
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:3464]:
```suggestion
.access-level-status {
  margin: unset;
}
```

File: ckanext/ontario_theme/templates/internal/snippets/search_form.html [hunk 2]
```diff
@@ -67,64 +68,77 @@ <h2 class="ontario-label">
...
-                  focusable="false"
-                  sol:category="primary"
-                  viewBox="0 0 24 24"
-                  preserveAspectRatio="xMidYMid meet">
-                <use xlink:href="#ontario-icon-close"></use>
-              </svg>
-            </span>
+      <div class="radio-buttons-status">
+        {% if '/dataset/' in current_url %}
+          <div class="access-level-status ontario-small-8 ontario-columns">
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/templates/internal/snippets/search_form.html:81]:
```suggestion
          <div class="ontario-callout access-level-status ontario-small-12 ontario-medium-8">
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 3]
```diff
@@ -490,15 +497,29 @@ input[type="reset"].ontario-search-reset:focus {
   display: none;
 }
 
+.radio-buttons-status{
+  order: 5;
+  background-color: white;
+  margin: 0px -20px;
+  padding: 33px 0px;
+}
+
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:506]:
```suggestion
.radio-buttons-status{
  order: 2;
  background-color: white;
  margin: 0 -16px;
  padding: 33px 0;
  display: flex;
  align-items: center;
}

```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 4]
```diff
@@ -401,6 +401,18 @@ input[type="reset"].ontario-search-reset:focus {
   padding-bottom: 16px;
 }
 
+.search-form .order-by-dropdown {
+  justify-content: center;
+  padding-top: 12px;
+}
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:407]:
```suggestion
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 5]
```diff
@@ -401,6 +401,18 @@ input[type="reset"].ontario-search-reset:focus {
...
+
+@media screen and (max-width: 40em) {
+  .access-level-status.ontario-small-8.ontario-columns { 
+    width: 90%;
+    margin-left: 1rem;
+  }
+}
+
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:415]:
```suggestion
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 6]
```diff
@@ -409,13 +421,10 @@ input[type="reset"].ontario-search-reset:focus {
   .search-form .control-order-by {
     display: flex;
     justify-content: end;
-    padding-top: 33px;
-
     margin-bottom: 0;
     background-color: #fff;
-    padding-right: 1rem;
+    padding: 5px;
     margin-left: -20px;
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:427]:
```suggestion
    padding-right: 1rem;
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 7]
```diff
@@ -490,15 +497,29 @@ input[type="reset"].ontario-search-reset:focus {
...
+.radio-buttons-status{
+  order: 5;
+  background-color: white;
+  margin: 0px -20px;
+  padding: 33px 0px;
+}
+
 @media (max-width: 767px) {
-  .search-form .control-order-by {
+  .search-form .order-by-dropdown {
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:508]:
```suggestion
  .search-form .control-order-by {
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 8]
```diff
@@ -529,6 +550,22 @@ input[type="reset"].ontario-search-reset:focus {
...
+  .access-level-status.ontario-small-8.ontario-columns { 
+    width: 55%;
+  }
+
+  .radio-buttons-status {
+  margin: 0px -20px;
+  padding: 33px 0px;
+  align-items: center;
+  }
+}
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:568]:
```suggestion
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 9]
```diff
@@ -490,15 +497,29 @@ input[type="reset"].ontario-search-reset:focus {
...
     display: block;
   }
 
+  .radio-buttons-status {
+    order: 6;
+    background-color: white;
+    margin: 0px -20px;
+    padding: 33px 0px;
+  }
+
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:522]:
```suggestion
  .radio-buttons-status {
    order: 6;
    margin: 0 -16px;
    padding: 33px 16px 0;
  }

```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 10]
```diff
@@ -409,13 +421,10 @@ input[type="reset"].ontario-search-reset:focus {
   .search-form .control-order-by {
     display: flex;
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:422]:
```suggestion
```

**Review by aleeexgreeen — CHANGES_REQUESTED** [review]
This is part one of my review to simplify the css changes. We have the DS callout component defined in our css, so it's best we use that, with our custom modifications. I'll tackle the js in the next part

**Comment by aleeexgreeen** [comment]
So the DS has the open and all levels colours listed as border-highlight classes for the callouts. We can add these two (see code chunk below) to the "Callouts - Design System" section of the css, the under review colour (`#92278f`) is already listed there. For the restricted colour, we can create our own colour following the same pattern as the others and the class starting with `ontario-border-highlights--<restricted colour name>`, this will be useful for what I recommend next. 

From here what I'll suggest we do is make use of the [data-attributes in html](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes). We can add attributes like `data-colour` and `data-display-name` to add both the corresponding colors and corresponding display name for each access level radio button. (Display name is defined in `access_level_radio_buttons.html`)
In the JavaScript, I recommend for the callout colour and text of the checked radio-button we add the class `ontario-border-highlight--{data-colour}` to the element that has the class `access_level_status` and to the `#access-level-sentence-value` we pass `{data-display-name}` to the innerHTML.

```
.ontario-border-highlight--teal {
  border-color: #367a76 !important
}

.ontario-border-highlight--taupe {
  border-color: #7b725c !important
}
```

Does this all make sense?

File: ckanext/ontario_theme/fanstatic/scripts/odc-access-level.js [hunk 11]
```diff
@@ -29,4 +29,17 @@
...
 
+  window.addEventListener('load', updateAccessLevelSentence)
+
+  function updateAccessLevelSentence() {
+    var selectedButton = document.querySelector('input[name="access_level"]:checked');
+    if (selectedButton) {
+      var displayName = selectedButton.getAttribute('data-display-name').trim();
+      var display_name_without_count = displayName.replace(/\s*\(\d+\)$/, '');
+      var calloutElement = document.querySelector('.access-level-status');
+      calloutElement.querySelector('#access-level-sentence-value').textContent = display_name_without_count;
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/scripts/odc-access-level.js:40]:
```suggestion
      var displayName = selectedButton.getAttribute('data-display-name').trim();
      var calloutElement = document.querySelector('.access-level-status');
      calloutElement.querySelector('#access-level-sentence-value').textContent = displayName;
```

File: ckanext/ontario_theme/fanstatic/common.css [hunk 12]
```diff
@@ -2287,7 +2307,19 @@ screen and (min-width: 96em) {
...
+  border-color: #367a76 !important;
+}
+
+.ontario-border-highlight--taupe {
+  border-color: #7b725c !important;
+}
+
+.ontario-border-highlight--red {
+  border-color: #D81A21 !important;
 }
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/fanstatic/common.css:2323]: Can you add a comment just above this rule to say it's specific to our theme? We plan on removing all design system code in favour for their npm package at some point, so we won't to lose this rule

File: ckanext/ontario_theme/templates/internal/snippets/access_level_radio_buttons.html [hunk 13]
```diff
@@ -29,9 +33,11 @@
...
         {% if "*" == request.params[name] %}
           {% set checked = "checked" %}
+          {% set display_name = _('All levels') %}
+          {% set all_levels_color = 'taupe' %}
         {% endif %}
         <div class="ontario-radios__item">
-          <input class="ontario-radios__input" id="radio-button-option-all" name="access_level" type="radio" value="*" {{ checked }}>
+          <input class="ontario-radios__input" id="radio-button-option-all" name="access_level" type="radio" value="*" data-color="{{ all_levels_color }}" data-display-name="{{ display_name }}" {{ checked }}>
           <label class="ontario-label ontario-radios__label"
                   for="radio-button-option-all">{{ _('All levels') }} ({{ h.localised_number(total_count) }})</label>
```
Comment (aleeexgreeen) [inline ckanext/ontario_theme/templates/internal/snippets/access_level_radio_buttons.html:42]: Can you change All levels to display_name, now that you have defined it 🙏🏿 ?

**Review by aleeexgreeen — CHANGES_REQUESTED** [review]
Great work! Just a few last changes and then we should be good to merge 👏🏿

File: ckanext/ontario_theme/fanstatic/scripts/odc-access-level.js (hunk 11 above)
Comment (KobikahChandran) [reply ckanext/ontario_theme/fanstatic/scripts/odc-access-level.js:40]: Shouldn't I keep line 38 to get rid of the count value?

Comment (aleeexgreeen) [reply ckanext/ontario_theme/fanstatic/scripts/odc-access-level.js:40]: data-display-name
In the HTML, the data-display-name attribute is just the name of the radio button with no count. So we won't need line 38

**Review by aleeexgreeen — APPROVED** [review]
All good! Great work @KobikahChandran !!

**Pull request merged**